tests/benchmark_sse1.py

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/env python
+# flake8: noqa
+# Copyright 2014 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+import tempfile, os, sys, shutil, json, re
+
+__rootpath__ = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+def path_from_root(*pathelems):
+  return os.path.join(__rootpath__, *pathelems)
+sys.path += [path_from_root('')]
+from tools import shared
+from tools.shared import PYTHON, WINDOWS, CLANG_CXX, EMCC, PIPE, V8_ENGINE
+from tools.shared import Popen
+
+
+temp_dir = tempfile.mkdtemp()
+
+# System info
+system_info = Popen([PYTHON, path_from_root('emrun'), '--system_info'], stdout=PIPE, stderr=PIPE).communicate()
+
+# Native info
+native_info = Popen(['clang', '-v'], stdout=PIPE, stderr=PIPE).communicate()
+
+# Emscripten info
+emscripten_info = Popen([PYTHON, EMCC, '-v'], stdout=PIPE, stderr=PIPE).communicate()
+
+# Run native build
+out_file = os.path.join(temp_dir, 'benchmark_sse1_native')
+if WINDOWS: out_file += '.exe'
+cmd = [CLANG_CXX] + shared.Building.get_native_building_args() + [path_from_root('tests', 'sse', 'benchmark_sse1.cpp'), '-O3', '-o', out_file]
+print 'Building native version of the benchmark:'
+print ' '.join(cmd)
+build = Popen(cmd, env=shared.Building.get_building_env(native=True))
+out = build.communicate()
+if build.returncode != 0:
+    sys.exit(1)
+
+native_results = Popen([out_file], stdout=PIPE, stderr=PIPE).communicate()
+print native_results[0]
+
+# Run emscripten build
+out_file = os.path.join(temp_dir, 'benchmark_sse1_html.js')
+cmd = [PYTHON, EMCC, path_from_root('tests', 'sse', 'benchmark_sse1.cpp'), '-O3', '-msse', '-s', 'SIMD=1', '-s', 'TOTAL_MEMORY=536870912', '-o', out_file]
+print 'Building Emscripten version of the benchmark:'
+print ' '.join(cmd)
+build = Popen(cmd)
+out = build.communicate()
+if build.returncode != 0:
+    sys.exit(1)
+
+cmd = V8_ENGINE + ['--experimental-wasm-simd', os.path.basename(out_file)]
+print ' '.join(cmd)
+old_dir = os.getcwd()
+os.chdir(os.path.dirname(out_file))
+wasm_results = Popen(cmd, stdout=PIPE, stderr=PIPE).communicate()
+os.chdir(old_dir)
+
+if not wasm_results:
+    raise Exception('Unable to run benchmark in V8!')
+
+if not wasm_results[0].strip():
+    print wasm_results[1]
+    sys.exit(1)
+
+print wasm_results[0]
+def strip_comments(text):
+    return re.sub('//.*?\n|/\*.*?\*/', '', text, re.S)
+benchmark_results = strip_comments(wasm_results[0])
+
+# Strip out unwanted print output.
+benchmark_results = benchmark_results[benchmark_results.find('{'):].strip()
+if '*************************' in benchmark_results:
+    benchmark_results = benchmark_results[:benchmark_results.find('*************************')].strip()
+
+print benchmark_results
```

Comment:
I thought we had switched to Python 3 already? Or is that just a WIP?

Reply:
This script is from old era, not Python 3 unfortunately. We are still dual Python2+Python3 in Emscripten. I'll have to see about converting this to be Python 3 friendly, although not critical for now.